crates/subspace-core-primitives/src/lib.rs

```diff
@@ -531,3 +532,66 @@ pub fn bidirectional_distance<T: num_traits::WrappingSub + Ord>(a: &T, b: &T) ->
     // Find smaller diff between 2 directions.
     diff.min(diff2)
 }
+
+pub use construct_uint::{PieceDistance, U256};
+
+#[allow(clippy::assign_op_pattern, clippy::ptr_offset_with_cast)]
+mod construct_uint {
+    //! This module is needed to scope clippy allows
+
+    use super::{bidirectional_distance, PieceIndexHash};
+    use num_traits::{WrappingAdd, WrappingSub};
+
+    uint::construct_uint! {
+        pub struct U256(4);
+    }
+
+    /// Distance to piece index hash from farmer identity
+    pub type PieceDistance = U256;
```

Comment:
The `PieceDistance` looks like farmer specific indeed. However, it cannot be easily moved because of the U256 impl-dependencies. The type needs a redesign leaving a plain U256 here and constructing some `PieceDistance` wrapper around it in the farmer. I can do it, no problem, however, it could violate @i1i1 's plan. What do you think, guys?

Reply:
I don't see any issues with leaving APIs here as is. There are strange things like `distance` method and `MIDDLE` constant, but we can live with them here as they are needed in many places across the protocol.